src/Connections/GitlabRepo.ts

```diff
@@ -739,26 +751,52 @@ ${data.description}`;
             comments = ` with ${result.commentCount} comments`;
         }
 
-        let approvalState = 'commented on';
+        let relation;
+        const discussionWithThread = result.discussions.find(discussionId => this.discussionThreads.has(discussionId));
+        if (discussionWithThread) {
+            const threadEventId = await this.discussionThreads.get(discussionWithThread)!.catch(_ => { /* already logged */ });
```

Comment:
The `!` seems like a bad smell. Can we do the `get` outside the if statement and check for it's undefined-ness?